x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -211,23 +228,26 @@ datasources:
         # buckets?
         - type: s3
           credential_profile_name: fb-aws
+          #shared_credential_file: /etc/filebeat/aws_credentials
           streams:
           - id?: {id}
-            metricset: s3
-            dataset: aws/s3
-            queue_url: https://sqs.myregion.amazonaws.com/123456/myqueue
+            dataset: aws.s3
+            queue_url: https://sqs.myregion.amazonaws.com/123456/sqs-queue
           - id?: {id}
-            metricset: s3access
-            dataset: aws/s3access
+            dataset: aws.s3access
```

Comment:
will we provide some mapping logic between dataset-metricset in case metricset is missing?

Reply:
This could become tricky as we would need a lookup. It is not always straight forward.

Reply:
So in this case explicity >= magic :)

Reply:
so we should always have metricset stated in case of metricbeat input

Reply:
Yes, we would say that **metricset** is indeed mandatory, without that we cannot pick up the right module when launching metricbeat and we configure the module with a custom option like `{metricset}.include_top_n`